hack/codegen.go

```diff
@@ -0,0 +1,734 @@
+	packageName := generatePackage(path)
+	// 7. check if go source file exists
+	goFile := filepath.Join(conformance, packageName, "feature.go")
+	isGoFileOk := files.Exists(goFile)
+
+	mapping := &Mapping{
+		Package:      packageName,
+		FeatureFile:  path,
+		Features:     featureSteps,
+		NewFunctions: featureSteps,
+		GoFile:       goFile,
+	}
+
+	// 8. Extract functions from go source code
+	if isGoFileOk {
+		goFunctions, err := extractFuncs(goFile)
+		if err != nil {
+			return fmt.Errorf("extracting go functions: %w", err)
+		}
+
+		mapping.GoDefinitions = goFunctions
+	}
+
+	// 9. check if feature file is in sync with go code
+	isInSync := false
+
+	signatureChanges := []SignatureChange{}
+
+	if isGoFileOk {
+		inFeatures := sets.NewString()
+		inGo := sets.NewString()
+
+		for _, feature := range mapping.Features {
+			inFeatures.Insert(feature.Name)
+		}
+
+		for _, gofunc := range mapping.GoDefinitions {
+			inGo.Insert(gofunc.Name)
+		}
+
+		if newFunctions := inFeatures.Difference(inGo); newFunctions.Len() > 0 {
+			log.Printf("Feature file %v contains %v new function/s", mapping.FeatureFile, newFunctions.Len())
+			isInSync = false
+
+			var funcs []Function
+			for _, f := range newFunctions.List() {
+				for _, feature := range mapping.Features {
+					if feature.Name == f {
+						funcs = append(funcs, feature)
+						break
+					}
+				}
+			}
+
+			mapping.NewFunctions = funcs
+		} else {
+			mapping.NewFunctions = []Function{}
+		}
+
+	FeaturesLoop:
+		for _, feature := range mapping.Features {
+			for _, gofunc := range mapping.GoDefinitions {
+				if feature.Name != gofunc.Name {
+					continue
+				}
+
+				// We need to compare function arguments checking only
+				// the number and type. Is not possible to rely in the name
+				// in the go code.
+				featKeys := feature.Args.Keys()
+				goKeys := gofunc.Args.Keys()
+				if len(featKeys) != len(goKeys) {
+					signatureChanges = append(signatureChanges, SignatureChange{
+						Function: gofunc.Name,
+						Have:     argsFromMap(gofunc.Args, true),
+						Want:     argsFromMap(feature.Args, true),
+					})
+
+					break FeaturesLoop
+				}
+
+				for index, k := range featKeys {
+					fv, _ := feature.Args.Get(k)
+					gv, _ := gofunc.Args.Get(goKeys[index])
+
+					if !reflect.DeepEqual(fv, gv) {
+						signatureChanges = append(signatureChanges, SignatureChange{
+							Function: gofunc.Name,
+							Have:     argsFromMap(gofunc.Args, true),
+							Want:     argsFromMap(feature.Args, true),
+						})
+
+						break FeaturesLoop
+					}
+				}
+
+			}
+		}
+	}
+
+	// 10. check signatures are ok
+	if len(signatureChanges) != 0 {
+		var argBuf bytes.Buffer
+		for _, sc := range signatureChanges {
+			argBuf.WriteString(fmt.Sprintf(`
+function %v
+	have %v
+	want %v
+`, sc.Function, sc.Have, sc.Want))
+		}
+
+		return fmt.Errorf("source file %v has a different signature/s:\n %v", mapping.GoFile, argBuf.String())
+	}
+
+	// 11. if in sync, nothing to do
+	if isInSync {
```

Comment:
I don't see any usage of `isInSync = true`. What's the intention here?

Reply:
fixed